storage/storage.cpp

```diff
@@ -1145,4 +1145,19 @@ void Storage::DoClickOnDownloadMap(TCountryId const & countryId)
   if (m_downloadMapOnTheMap)
     m_downloadMapOnTheMap(countryId);
 }
+
+void Storage::ForEachInSubtree(TCountryId const & root, TForEachFunction const & toDo) const
```

Comment:
Not sure that toDo should be passed by a const-reference only. Imagine a functor with a mutable state, for example a functor that collects names of all nodes in a vector - it can't be passed by a const reference, right?